webapp/src/components/harbour/QuotaOverview.tsx

```diff
@@ -0,0 +1,96 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useConnectWallet } from "@web3-onboard/react";
+import { getAddress } from "ethers";
+import { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+import { Box } from "@/components/Groups";
+import type { SettingsFormData } from "@/components/settings/SettingsForm";
+import { useHarbourRpcProvider } from "@/hooks/useRpcProvider";
+import { checkedAddressSchema } from "@/lib/validators";
+import { FormItem, SubmitItem } from "../Forms";
+import { QuotaStats } from "./QuotaStats";
+import { QuotaTokenBalance } from "./QuotaTokenBalance";
+
+const createQuotaStatsFormSchema = () =>
+	z.object({
+		signerAddress: z.union([checkedAddressSchema, z.literal("")]),
+	});
+
+type QuotaStatsFormData = z.infer<
+	ReturnType<typeof createQuotaStatsFormSchema>
+>;
+
+function QuotaOverview({
+	currentSettings,
+	className,
+}: {
+	currentSettings: Partial<SettingsFormData>;
+	className: string;
+}) {
+	const [{ wallet }] = useConnectWallet();
+	const [signerAddress, setSignerAddress] = useState<string | undefined>();
+	const [refreshCounter, setRefreshCounter] = useState<number>(0);
+	const [isLoadingQuota, setIsLoadingQuota] = useState<boolean>(false);
+	const { provider: harbourProvider } = useHarbourRpcProvider();
+
+	const {
+		register,
+		handleSubmit,
+		formState: { errors },
+		reset,
+	} = useForm<QuotaStatsFormData>({
+		resolver: zodResolver(createQuotaStatsFormSchema()),
```

Comment:
Ah this is cool - it lets you have typed guarantees on the form data (instead of hand-rolling validation code on the various input items into TypeScript types)? If so - added to my mental list of things to use in my React projects :P

Reply:
yeah, Misha introduced this, seems nice :D